### PR TITLE
new data source for MongoDB instances

### PR DESCRIPTION
**Pull request by malferov**
new data source for MongoDB instances and its documentation

### CODE REVIEW

File: alicloud/data_source_alicloud_mongo_instances.go [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+				}
+			}
+			dbi = append(dbi, item)
+		}
+
+		if len(resp.DBInstances.DBInstance) < PageSizeLarge {
+			break
+		}
+
+		args.PageNumber = args.PageNumber + requests.NewInteger(1)
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_mongo_instances.go]: HI @malferov line 206 is a bug and you can refer to https://github.com/terraform-providers/terraform-provider-alicloud/pull/250/commits/7019687b9931070c00a0281d4f39371990729b26#diff-90d96cd66a3012b469997eb1cbceb593R132

File: alicloud/services/dds/client.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
+package dds
```
Comment (xiaozhu36) [inline alicloud/services/dds/client.go]: I will push mongoDB developer to publish official go sdk. Please be patient.

File: alicloud/data_source_alicloud_mongo_instances.go (hunk 1 above)
Comment (malferov) [reply alicloud/data_source_alicloud_mongo_instances.go]: fixed

File: alicloud/data_source_alicloud_mongo_instances.go [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+				ValidateFunc: validateNameRegex,
+			},
+			"engine": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validateAllowedStringValue([]string{
+					"MongoDB",
+				}),
+			},
+			"db_type": {
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_mongo_instances.go]: I think line 30 should be instance_type

File: alicloud/data_source_alicloud_mongo_instances.go [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"charge_type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"db_type": {
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_mongo_instances.go]: what is db_type

File: alicloud/data_source_alicloud_mongo_instances.go [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+										Computed: true,
+									},
+									"class": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+								},
+							},
+						},
+						"shard": {
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_mongo_instances.go]: I think line 129 should be shards

File: alicloud/services/dds/client.go (hunk 2 above)
Comment (xiaozhu36) [reply alicloud/services/dds/client.go]: Please use official go sdk: https://github.com/aliyun/alibaba-cloud-sdk-go/tree/master/services/dds

**Comment by malferov** [comment]
@xiaozhu36 updated and switched to official go sdk. please review

File: alicloud/data_source_alicloud_mongo_instances.go [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+	return &schema.Resource{
+		Read: dataSourceAlicloudMongoInstancesRead,
+
+		Schema: map[string]*schema.Schema{
+			"name_regex": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateNameRegex,
+			},
+			"engine": {
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_mongo_instances.go]: If engine only has one validate value, the field can be removed and new flied 'availability_zone' and 'instance_class' can be added.

**Comment by xiaozhu36** [comment]
@malferov Thanks for your contribution. Can you rebase all of commits and remove useless annotation before the next submitting?

**Comment by malferov** [comment]
@xiaozhu36 indeed `engine` argument is useless due to one possible value so far. so i have removed that and added two more according your advice above. the branch has been rebased. thanks

**Pull request merged**